src/uwtools/drivers/driver.py

```diff
@@ -53,22 +60,36 @@ def requirements(self) -> None:
         """
 
     @abstractmethod
-    def resources(self, platform: dict) -> Mapping:
+    def resources(self) -> Mapping:
         """
         Parses the config and returns a formatted dictionary for the batch script.
         """
 
     @abstractmethod
-    def run(self) -> None:
+    def run(self, cycle: datetime) -> None:
         """
         Run the NWP tool.
+
+        :param cycle: The time stamp of the current cycle to run.
         """
 
-    @abstractmethod
-    def run_cmd(self, *args, run_cmd: str, exec_name: str) -> str:
+    def run_cmd(self, *args) -> str:
         """
         The command-line command to run the NWP tool.
+
+        :param args: Any number of native flags for the run command
+        """
+        run_cmd = self._platform_config["mpicmd"]
+        exec_name = self._config["exec_name"]
+        args_str = " ".join(str(arg) for arg in args)
```

Comment:
We're currently requiring the caller to pass `args` in, but where does the caller get them? I guess ultimately the answer has to be "from the config file" (or, maybe, optionally, and/or from key=value command-line pairs and/or environment variables), and if that's the case -- and if we don't need those values elsewhere -- it probably makes sense to obtain the values as close to the use site as possible (e.g. in this method) instead of passing them as arguments.

I say "if we don't need those values elsewhere" because I think we should avoid doing the same `dict` lookups in multiple places whenever we can, because that will require multiple code changes when something in the config changes, which I think is very likely going to happen, frequently, as our understanding of what UW YAML should be evolves. If we don't keep things DRY, I'm afraid we'll start resisting making changes that we should make because the changes will be painful to implement. Things look pretty DRY so far, but we're kind of just getting started, so maybe it's worth being watchful.

Reply:
Cleaning this up actually ends up DRYing it out quite a bit!